pkg/controller/garbagecollector/graph_builder.go

```diff
@@ -248,17 +253,21 @@ func (gb *GraphBuilder) startMonitors() {
 	// that they don't get unexpected events on their work queues.
 	<-gb.informersStarted
 
+	gb.sharedInformers.Start(gb.stopCh)
 	monitors := gb.monitors
-	started := 0
-	for _, monitor := range monitors {
-		if monitor.stopCh == nil {
-			monitor.stopCh = make(chan struct{})
-			gb.sharedInformers.Start(gb.stopCh)
-			go monitor.Run()
-			started++
+	for gvr := range monitors {
+		// TODO(kdelga): Should we only be doing this when ok?
+		if doneCh, ok := gb.sharedInformers.DoneChannelFor(gvr); ok {
+			go func() {
+				<-doneCh
```

Comment:
comment on what happens if we happen to sync between receiving a signal on this channel and acquiring the lock

Reply:
Comment added to the `addStoppedResources` method signature that the resource will wait until the following sync to restart its informer. Let me know if that is clear (and correct) or if you were thinking differently.